manager/integration/tests/test_basic.py

```diff
@@ -3402,7 +3402,6 @@ def test_allow_volume_creation_with_degraded_availability_restore(set_random_bac
     common.wait_for_replica_scheduled(client, dst_vol_name,
```

Comment:
This step is while the volume is attached, so it should be able to verify `expect_fail=2`, right?

Reply:
Found, I think the problem is at `wait_for_volume_condition_scheduled`.
`status` would only be `true` after the volume are detached because there are two replicas unscheduled.

Reply:
> Found, I think the problem is at `wait_for_volume_condition_scheduled`. `status` would only be `true` after the volume are detached because there are two replicas unscheduled.

@mantissahz could you elaborate this a bit? I assume the status should be true when the volume is degraded when the replicas are scheduled during the attachment, right?

Reply:
ref: https://github.com/longhorn/longhorn-manager/blob/5a75273b50210cc5e652efc7597d821aac65702a/controller/volume_controller.go#L1169-L1169

Not sure about the previous context, but I feel this condition is a little confusing here because even if the volume is degraded, the condition of `scheduled` should be true as well but w/ a meaningful reason to explain that only some replicas are actually scheduled. 

Probably you can check `Robustness` of volume to see if it's `degraded`, so can you add `wait_for_volume_degraded` here instead of just removing the check?